### PR TITLE
Avoid blocking non-reproducible randomness in test

### PR DESCRIPTION
**Pull request by jasontedor**
The security documentation test uses SecureRandom#getStrongInstance. This defaults to securerandom.strongAlgorithms=NativePRNGBlocking:SUN,DRBG:SUN which means a blocking implementation that reads from /dev/random. This means that this test can stall if the entropy on the machine is exhausted. Anyway, it also means that the randomness is non-reproducible, a thing that we try to avoid in tests. This commit switches to a boring randomness source to avoid the blocking, and to keep the test reproducible.

### CODE REVIEW

**Review by jaymode — APPROVED** [review]
LGTM

**Comment by jaymode** [comment]
> strong-bad-how-do-you-type-with-boxing-gloves-on

I JUST MASH KEYS UNTIL IT MAKES SENSE 😛

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public void testPutUser() throws Exception {
         }
         {
             byte[] salt = new byte[32];
-            SecureRandom.getInstanceStrong().nextBytes(salt);
+            random().nextBytes(salt);
```
Comment (jaymode) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java]: I didn't think about this on first look, but can you add a comment that this test intentionally uses a reproducible random and not a securerandom

Comment (jasontedor) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java]: I pushed 16eceac5403a37e3df0ced0ff84ec4afda61966c.

**Comment by jasontedor** [comment]
Thanks @jaymode!

**Pull request merged**